service/route_handlers.go

```diff
@@ -536,12 +537,15 @@ func HandleSamlAssertion(w http.ResponseWriter, r *http.Request, assertion *saml
 		redirectBackBaseValue = server.GetRancherAPIHost()
 	}
 
+	redirectURL := server.GetSamlRedirectURL(redirectBackBaseValue, redirectBackPathValue)
+
 	if !isWhitelisted(redirectBackBaseValue, serviceProvider.RedirectWhitelist) {
 		log.Errorf("Cannot redirect to anything other than whitelisted domains and rancher api host")
+		redirectURL = addErrorToRedirect(redirectURL, "403")
+		http.Redirect(w, r, redirectURL, http.StatusFound)
```

Comment:
I think we should redirect to GetRancherAPIHost() with a "403" in this case. We should not redirect to where the user is asking us to go.